src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerArgumentBinder.cs

```diff
@@ -111,10 +132,37 @@ public class ControllerArgumentBinder : IControllerActionArgumentBinder
                 throw new ArgumentNullException(nameof(controllerContext));
             }
 
+            var valueProvider = await CompositeValueProvider.CreateAsync(
+                controllerContext,
+                controllerContext.ValueProviderFactories);
+
+            return await BindModelAsync(parameter, controllerContext, valueProvider);
+        }
+
+        public async Task<ModelBindingResult?> BindModelAsync(
+            ParameterDescriptor parameter,
+            ControllerContext controllerContext,
+            IValueProvider valueProvider)
```

Comment:
Sure but why is the value provider created at all in the case where all parameters have `[FromBody]` or similar? I was expecting a lazy-init property on the `ModelBindingContext` that's backed in the `ControllerContext`.